tests/components/switchbot/test_cover.py

```diff
@@ -0,0 +1,315 @@
+    SERVICE_CLOSE_COVER_TILT,
+    SERVICE_OPEN_COVER,
+    SERVICE_OPEN_COVER_TILT,
+    SERVICE_SET_COVER_POSITION,
+    SERVICE_SET_COVER_TILT_POSITION,
+    SERVICE_STOP_COVER,
+    SERVICE_STOP_COVER_TILT,
+)
+from homeassistant.core import HomeAssistant, State
+
+from . import WOBLINDTILT_SERVICE_INFO, WOCURTAIN3_SERVICE_INFO, make_advertisement
+
+from tests.common import MockConfigEntry, mock_restore_cache
+from tests.components.bluetooth import inject_bluetooth_service_info
+
+
+async def test_curtain3_setup(
+    hass: HomeAssistant, mock_entry_factory: Callable[[str], MockConfigEntry]
+) -> None:
+    """Test setting up the Curtain3."""
+    inject_bluetooth_service_info(hass, WOCURTAIN3_SERVICE_INFO)
+
+    entry = mock_entry_factory(sensor_type="curtain")
+
+    entity_id = "cover.test_name"
+    mock_restore_cache(
+        hass,
+        [
+            State(
+                entity_id,
+                CoverState.OPEN,
+                {ATTR_CURRENT_POSITION: 50},
+            )
+        ],
+    )
+
+    entry.add_to_hass(hass)
+    assert await hass.config_entries.async_setup(entry.entry_id)
+    await hass.async_block_till_done()
+
+    assert hass.states.get(entity_id).state == CoverState.OPEN
+    assert hass.states.get(entity_id).attributes[ATTR_CURRENT_POSITION] == 50
+
+
+async def test_curtain3_controlling(
+    hass: HomeAssistant, mock_entry_factory: Callable[[str], MockConfigEntry]
+) -> None:
+    """Test Curtain3 controlling."""
+    inject_bluetooth_service_info(hass, WOCURTAIN3_SERVICE_INFO)
+
+    entry = mock_entry_factory(sensor_type="curtain")
+    entry.add_to_hass(hass)
+
+    with (
+        patch(
+            "switchbot.SwitchbotCurtain.open", new=AsyncMock(return_value=True)
+        ) as mock_open,
+        patch(
+            "switchbot.SwitchbotCurtain.close", new=AsyncMock(return_value=True)
+        ) as mock_close,
+        patch(
+            "switchbot.SwitchbotCurtain.stop", new=AsyncMock(return_value=True)
+        ) as mock_stop,
+        patch(
+            "switchbot.SwitchbotCurtain.set_position", new=AsyncMock(return_value=True)
+        ) as mock_set_position,
+    ):
+        assert await hass.config_entries.async_setup(entry.entry_id)
+        await hass.async_block_till_done()
+
+        entity_id = "cover.test_name"
+        address = "AA:BB:CC:DD:EE:FF"
+        service_data = b"{\xc06\x00\x11D"
+
+        # Test open
+        manufacturer_data = b"\xcf;Zwu\x0c\x19\x0b\x05\x11D\x006"
+        await hass.services.async_call(
+            COVER_DOMAIN, SERVICE_OPEN_COVER, {ATTR_ENTITY_ID: entity_id}, blocking=True
+        )
+        inject_bluetooth_service_info(
+            hass, make_advertisement(address, manufacturer_data, service_data)
+        )
+        await hass.async_block_till_done()
+
+        mock_open.assert_awaited_once()
+        assert hass.states.get(entity_id).state == CoverState.OPEN
+        assert hass.states.get(entity_id).attributes[ATTR_CURRENT_POSITION] == 95
```

Comment:
```suggestion
        assert hass.states[entity_id].state == CoverState.OPEN
        assert hass.states[entity_id].attributes[ATTR_CURRENT_POSITION] == 95
```

Please use `[]` instead of `get()` as it would be an error for it to not be there

Reply:
Nevermind, I'm reading it as a dict fetch but its a fetch from the state machine which is using the `get` api